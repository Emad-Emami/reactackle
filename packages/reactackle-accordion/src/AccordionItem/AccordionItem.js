import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { noop, withTheme } from 'reactackle-core';

import {
  HeadingStyled,
  TitleBoxStyled,
  TitleContentStyled,
  IconExpandWrapperStyled,
  IconWrapperStyled,
  TitleStyled,
  ContentStyled,
} from './styles';

const propTypes = {
  /*
   * Item id
   */
  id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /*
   * Item title
   */
  title: PropTypes.string,
  /*
   * Show/hide item's content
   */
  expanded: PropTypes.bool,
  /*
   * Removes content paddings
   */
  contentBlank: PropTypes.bool,
  /*
   * Adds additional content slot to the item's header
   */
  headerSlot: PropTypes.node,
  /*
   * Function to be called on title click
   */
  onToggleExpanded: PropTypes.func,
  // eslint-disable-next-line react/require-default-props
  theme: PropTypes.object,
};

const defaultProps = {
  id: null,
  title: '',
  expanded: false,
  contentBlank: false,
  onToggleExpanded: noop,
  headerSlot: null,
};

class AccordionItem extends Component {
  constructor(props) {
    super(props);
    this._handleClick = this._handleClick.bind(this);
  }
  
  _handleClick() {
    this.props.onToggleExpanded({ id: this.props.id });
  }
  
  render() {
    const {
      expanded,
      contentBlank,
    } = this.props;
    
    const title = this.props.title &&
      <TitleStyled expanded={expanded}>{this.props.title}</TitleStyled>;

    const styleKey = expanded ? 'expanded' : 'collapsed';
    
    return [
      <HeadingStyled key="heading" role="heading">
        <TitleBoxStyled
          expanded={expanded}
          onClick={this._handleClick}
          role="button"
          aria-expanded={expanded}
          aria-controls={this.props.id}
          id={`labelfor${this.props.id}`}
        >
          <TitleContentStyled>
            {title}
            {this.props.headerSlot}
          </TitleContentStyled>

          <IconExpandWrapperStyled>
            <IconWrapperStyled expanded={expanded}>
              {this.props.theme.reactackle.components.accordion.item
                .expandIcon[styleKey].iconElement}
            </IconWrapperStyled>
          </IconExpandWrapperStyled>
        </TitleBoxStyled>
      </HeadingStyled>,
      <ContentStyled
        key="content"
        expanded={expanded}
        contentBlank={contentBlank}
        id={this.props.id}
        role="region"
        aria-labelledby={`labelfor${this.props.id}`}
      >
        {this.props.children}
      </ContentStyled>,
    ];
  }
}

AccordionItem.propTypes = propTypes;
AccordionItem.defaultProps = defaultProps;
AccordionItem.displayName = 'AccordionItem';

export default withTheme(AccordionItem);
