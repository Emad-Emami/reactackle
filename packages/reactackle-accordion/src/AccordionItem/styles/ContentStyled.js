import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import {
  extractThemeOrDefault,
  getValueString,
  transition,
} from 'reactackle-core';

const propTypes = {
  contentBlank: PropTypes.bool,
  expanded: PropTypes.bool,
};

const defaultProps = {
  expanded: false,
  contentBlank: false,
};

const size = ({ theme: themeFromProvider, contentBlank }) => {
  const theme = extractThemeOrDefault(themeFromProvider);

  const {
    paddingY,
    paddingX,
  } = theme.reactackle.components.accordion.item.content;

  return !contentBlank && css`
    padding: ${getValueString(paddingY)} ${getValueString(paddingX)};
  `;
};

const expanded = ({ theme: themeFromProvider, expanded }) => {
  const theme = extractThemeOrDefault(themeFromProvider);

  return expanded
    ? `
      pointer-events: auto;
      opacity: 1;
      position: relative;
      z-index: 1;
      transform: translateY(0);
    `
    : css`
      transform: translateY(${getValueString(
        theme.reactackle.components.accordion.item.content.animatedOffset,
      )});
      pointer-events: none;
      opacity: 0;
      position: fixed;
      z-index: -1;
    `;
};

export const ContentStyled = styled.dd`
  margin: 0;
  ${size}
  ${expanded}
  ${transition('transform')}
`;

ContentStyled.propTypes = propTypes;
ContentStyled.defaultProps = defaultProps;
