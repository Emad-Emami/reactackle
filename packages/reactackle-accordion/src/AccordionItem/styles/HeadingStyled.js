import styled from 'styled-components';

export const HeadingStyled = styled.dt`
  position: relative;
  z-index: 2;
`;
