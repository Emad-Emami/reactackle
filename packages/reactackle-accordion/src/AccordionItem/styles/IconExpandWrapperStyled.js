import styled, { css } from 'styled-components';
import {
  extractThemeOrDefault,
  getValueString,
} from 'reactackle-core';

const size = ({ theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const path = theme.reactackle.components.accordion.item;

  const {
    height,
    width,
  } = theme.reactackle.components.accordion.item.expandIcon;

  return css`
    min-height: ${getValueString(path.title.minHeight)};
    width: ${getValueString(width || height)};
  `;
};

export const IconExpandWrapperStyled = styled.div`
  display: flex;
  align-items: center;
  ${size}
`;

IconExpandWrapperStyled.displayName = 'IconExpandWrapperStyled';
