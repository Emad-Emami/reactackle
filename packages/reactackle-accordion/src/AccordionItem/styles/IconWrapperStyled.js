import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import {
  extractThemeOrDefault,
  getValueString,
  transition,
  iconSizeMixin,
} from 'reactackle-core';

const propTypes = {
  expanded: PropTypes.bool,
};

const defaultProps = {
  expanded: false,
};

const expanded = ({ theme: themeFromProvider, expanded }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const styleKey = expanded ? 'expanded' : 'collapsed';

  const {
    opacity,
    rotateDegree,
  } = theme.reactackle.components.accordion.item.expandIcon[styleKey];

  return css`
    opacity: ${opacity};
    ${rotateDegree !== null && `transform: rotate(${rotateDegree});`}
  `;
};

const size = ({ theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);

  const {
    height,
    width,
    imgHeight,
    imgWidth,
  } = theme.reactackle.components.accordion.item.expandIcon;

  return css`
    ${iconSizeMixin(
      getValueString(height),
      getValueString(imgHeight || height),
      getValueString(width || height),
      getValueString(imgWidth || imgHeight || height),
    )}
  `;
};

export const IconWrapperStyled = styled.div`
  display: flex;
  align-items: center;
  ${expanded}
  ${size}
  ${transition('opacity, transform')}
`;

IconWrapperStyled.propTypes = propTypes;
IconWrapperStyled.defaultProps = defaultProps;
