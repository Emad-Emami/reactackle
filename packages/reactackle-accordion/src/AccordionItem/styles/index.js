export * from './HeadingStyled';
export * from './TitleBoxStyled';
export * from './TitleContentStyled';
export * from './TitleStyled';
export * from './IconExpandWrapperStyled';
export * from './IconWrapperStyled';
export * from './ContentStyled';
