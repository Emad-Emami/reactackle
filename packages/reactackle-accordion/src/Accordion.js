import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { noop, getKey, registerDefaultComponentTheme } from 'reactackle-core';
import AccordionItem from './AccordionItem/AccordionItem';
import { AccordionStyled } from './styles/AccordionStyled';

import componentTheme from './styles/theme';

registerDefaultComponentTheme('accordion', componentTheme);

const propTypes = {
  /**
   * Speed of exit animation
   */
  items:
    PropTypes.arrayOf(
        PropTypes.shape({
          ...AccordionItem.propTypes,
          content: PropTypes.node,
        }),
    ),
  /**
   * Define if Item has state itself
   */
  stateless: PropTypes.bool,
  /**
   * Array of expanded Item's ids
   */
  expandedItemIds: PropTypes.arrayOf(PropTypes.string),
  /**
   * Expand all items at once
   */
  expandAll: PropTypes.bool,
  /**
   * If true, only one item may be expanded at one time
   */
  single: PropTypes.bool,
  /**
   * Function to be called on Item title click
   */
  onChange: PropTypes.func,
};

const defaultProps = {
  items: [],
  expandedItemIds: [],
  expandAll: false,
  single: false,
  stateless: false,
  onChange: noop,
};

export default class Accordion extends Component {
  constructor(props) {
    super(props);

    if (!props.stateless) {
      let expandedItemIds;
      if (props.expandAll)
        expandedItemIds = props.items.map(item => item.id);
      else
        expandedItemIds = props.expandedItemIds;

      this.state = {
        expandedItemIds,
      };
    }
    this._handleItemExpand = this._handleItemExpand.bind(this);
  }

  componentWillReceiveProps({ expandedItemIds }) {
    if (
      expandedItemIds !== this.props.expandedItemIds
      && !this.props.stateless
    ) this.setState({ expandedItemIds });
  }

  _getExpandedItemIds() {
    if (!this.props.stateless)
      return this.state.expandedItemIds;

    return this.props.expandedItemIds;
  }

  _toggleItemExpand(itemId) {
    let expandedItemIds = this._getExpandedItemIds().slice();
    const idx = expandedItemIds.indexOf(itemId);

    if (expandedItemIds.length && expandedItemIds.includes(itemId))
      expandedItemIds.splice(idx, 1);
    else if (this.props.single)
      expandedItemIds = [itemId];
    else
      expandedItemIds.push(itemId);

    return expandedItemIds;
  }

  _isExpanded(itemId) {
    if (!this.props.stateless)
      return this.state.expandedItemIds.includes(itemId);

    return this.props.expandedItemIds.includes(itemId);
  }

  _handleItemExpand({ id }) {
    const newExpandedIds = this._toggleItemExpand(id);

    if (!this.props.stateless) {
      this.setState({
        expandedItemIds: newExpandedIds,
      });
    }
    this.props.onChange({
      expandedItemIds: newExpandedIds,
    });
  }

  render() {
    const items = this.props.items && this.props.items.map(
      (item, index) => {
        const expanded = this._isExpanded(item.id);

        return (
          <AccordionItem
            key={getKey(item, index)}
            id={item.id}
            title={item.title}
            expanded={expanded}
            headerSlot={item.headerSlot}
            onToggleExpanded={this._handleItemExpand}
            {...item}
          >
            {item.content}
          </AccordionItem>
        );
      },
    );

    return (
      <AccordionStyled role="presentation">
        {items}
        {this.props.children}
      </AccordionStyled>
    );
  }
}

Accordion.propTypes = propTypes;
Accordion.defaultProps = defaultProps;
Accordion.displayName = 'Accordion';
