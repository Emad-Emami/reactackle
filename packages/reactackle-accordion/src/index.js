import Accordion from './Accordion';
import AccordionItem from './AccordionItem/AccordionItem';

export { Accordion, AccordionItem };
