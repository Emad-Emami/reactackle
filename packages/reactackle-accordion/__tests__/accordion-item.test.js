import React from 'react';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { AccordionItem } from '../src';
import { TitleBoxStyled } from '../src/AccordionItem/styles';

describe('<AccordionItem />', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <AccordionItem />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with contentBlank prop set to true', () => {
    const tree = renderer.create(
      <AccordionItem contentBlank />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with expanded prop set to true', () => {
    const tree = renderer.create(
      <AccordionItem expanded />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with expanded id to string', () => {
    const tree = renderer.create(
      <AccordionItem id='testId' />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with expanded id to number', () => {
    const tree = renderer.create(
      <AccordionItem id={1} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with title to string', () => {
    const tree = renderer.create(
      <AccordionItem title='hello' />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('one call onToggleExpanded event', () => {
    const mockFn = jest.fn(),
      wrapper = mount(
        <AccordionItem onToggleExpanded={mockFn} />,
      );

      wrapper.find(TitleBoxStyled).simulate('click');

      expect(toJson(wrapper)).toMatchSnapshot();
      expect(mockFn).toHaveBeenCalledTimes(1);
  });
});