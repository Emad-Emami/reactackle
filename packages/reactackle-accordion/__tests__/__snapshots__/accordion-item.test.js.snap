// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionItem /> one call onToggleExpanded event 1`] = `
<WithTheme(AccordionItem)
  contentBlank={false}
  expanded={false}
  headerSlot={null}
  id={null}
  onToggleExpanded={[Function]}
  title=""
>
  <AccordionItem
    contentBlank={false}
    expanded={false}
    headerSlot={null}
    id={null}
    onToggleExpanded={[Function]}
    theme={
      Object {
        "reactackle": Object {
          "baseModule": 8,
          "body": Object {
            "backgroundColor": "#ffffff",
            "fontColor": "#212121",
            "fontColorAlt": "#ffffff",
            "fontFamily": "sansSerif",
            "fontWeight": "400",
            "lineHeight": 1.5,
          },
          "breakpoints": Object {
            "large": "80em",
            "medium": "60em",
            "small": "30em",
            "xlarge": "90em",
            "xsmall": "0",
            "xxlarge": "120em",
          },
          "breakpointsEmBase": 16,
          "color": Object {
            "alert": "#f04124",
            "alertFgTextColor": "#ffffff",
            "black": "#000000",
            "error": "#f04124",
            "errorFgTextColor": "#ffffff",
            "info": "#d9edf7",
            "infoFgTextColor": "#212121",
            "main": "#2196F3",
            "mainFgTextColor": "#ffffff",
            "secondary": "#673AB7",
            "secondaryFgTextColor": "#ffffff",
            "success": "#43AC6A",
            "successFgTextColor": "#ffffff",
            "transparent": "hsla(0, 0%, 100%, 0)",
            "warning": "#f0b900",
            "warningFgTextColor": "#212121",
            "white": "#ffffff",
          },
          "colorAnchor": "#208dcc",
          "colorBorder": "#e0e0e0",
          "colorBrand": Object {
            "facebook": "#3b5998",
            "googlePlus": "#dd4b39",
            "vkontakte": "#45668e",
          },
          "colorPaletteGrey": Object {
            "100": "#f5f5f5",
            "200": "#eeeeee",
            "300": "#e0e0e0",
            "400": "#bdbdbd",
            "50": "#fafafa",
            "500": "#9e9e9e",
            "600": "#757575",
            "700": "#616161",
            "75": "#f7f7f7",
            "800": "#424242",
            "900": "#212121",
          },
          "components": Object {
            "accordion": Object {
              "item": Object {
                "content": Object {
                  "animatedOffset": -20,
                  "paddingX": 16,
                  "paddingY": 12,
                },
                "expandIcon": Object {
                  "collapsed": Object {
                    "iconElement": <IconArrowChevronRight
                      color="currentColor"
                      size="custom"
      />,
                    "opacity": 0.7,
                    "rotateDegree": 0,
                  },
                  "expanded": Object {
                    "iconElement": <IconArrowChevronRight
                      color="currentColor"
                      size="custom"
      />,
                    "opacity": 1,
                    "rotateDegree": "90deg",
                  },
                  "height": 24,
                  "imgHeight": 20,
                  "imgWidth": null,
                  "width": null,
                },
                "title": Object {
                  "align": "top",
                  "collapsed": Object {
                    "backgroundColor": "#ffffff",
                    "color": "#212121",
                    "focus": Object {
                      "backgroundColor": "#f7f7f7",
                      "opacity": 1,
                    },
                    "fontSize": 12,
                    "fontWeight": "400",
                    "hover": Object {
                      "backgroundColor": "#fafafa",
                      "opacity": 1,
                    },
                    "lineHeight": 1.5,
                    "opacity": 0.7,
                    "paddingBottom": 8,
                    "paddingLeft": 16,
                    "paddingTop": 8,
                    "textTransform": "uppercase",
                  },
                  "expanded": Object {
                    "backgroundColor": "#ffffff",
                    "color": "#212121",
                    "focus": Object {
                      "backgroundColor": "#f7f7f7",
                      "opacity": 1,
                    },
                    "fontSize": 12,
                    "fontWeight": "600",
                    "hover": Object {
                      "backgroundColor": "#fafafa",
                      "opacity": 1,
                    },
                    "lineHeight": 1.5,
                    "opacity": 1,
                    "paddingBottom": 8,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 8,
                    "textTransform": "uppercase",
                  },
                  "minHeight": 34,
                },
              },
              "separator": Object {
                "color": "#e0e0e0",
                "style": "solid",
                "thickness": 1,
              },
            },
          },
          "contentWidth": 1360,
          "direction": Object {
            "float": "left",
            "text": "ltr",
          },
          "fontColor": Object {
            "light": "#e0e0e0",
            "medium": "#9e9e9e",
          },
          "fontFamily": Object {
            "monospace": "Consolas, Liberation Mono, Courier, monospace",
            "sansSerif": "Open Sans, Helvetica Neue, Helvetica, Roboto, Arial, sans-serif",
            "serif": "Georgia, Cambria, Times New Roman, Times, serif",
          },
          "fontSize": Object {
            "-1": 12,
            "-2": 10,
            "0": 13,
            "1": 15,
            "2": 18,
            "3": 24,
            "4": 34,
            "5": 45,
            "6": 56,
            "7": 112,
          },
          "fontWeight": Object {
            "bold": "700",
            "light": "400",
            "normal": "400",
            "semibold": "600",
          },
          "radiusDefault": 2,
          "radiusRounded": 1000,
          "totalColumns": 12,
          "zIndex": Object {
            "alertDesktop": 9000,
            "alertMobile": 5000,
            "base": 1,
            "dropdown": 7000,
            "modal": 8500,
            "moon": 9999,
            "nav": 500,
            "overlay": 8000,
            "popup": 8700,
            "tooltip": 9990,
          },
        },
      }
    }
    title=""
  >
    <HeadingStyled
      key="heading"
      role="heading"
    >
      <dt
        className="HeadingStyled-jcCngC diMfOQ"
        role="heading"
      >
        <TitleBoxStyled
          aria-controls={null}
          aria-expanded={false}
          expanded={false}
          id="labelfornull"
          onClick={[Function]}
          role="button"
        >
          <button
            aria-controls={null}
            aria-expanded={false}
            className="TitleBoxStyled-QGbRI bwRMmk"
            id="labelfornull"
            onClick={[Function]}
            role="button"
          >
            <TitleContentStyled>
              <div
                className="TitleContentStyled-eVpLoT bzPKhr"
              />
            </TitleContentStyled>
            <IconExpandWrapperStyled>
              <div
                className="IconExpandWrapperStyled-jsDvvw cHZrOx"
              >
                <IconWrapperStyled
                  expanded={false}
                >
                  <div
                    className="IconWrapperStyled-hDFZgP iRWScr"
                  >
                    <IconArrowChevronRight
                      color="currentColor"
                      size="custom"
                    >
                      <IconSvg
                        backgroundColor=""
                        border={false}
                        borderWidth={0}
                        color="currentColor"
                        colorScheme="default"
                        flip="none"
                        rotate={0}
                        rounded={false}
                        size="custom"
                      >
                        <IconStyled
                          backgroundColor=""
                          border={false}
                          borderWidth={0}
                          color=""
                          colorScheme="default"
                          customColor="currentColor"
                          flip="none"
                          rotate={0}
                          rounded={false}
                          size="normal"
                          sizeKey="custom"
                        >
                          <div
                            className="IconSvgStyled-cegYAl gxFYyV"
                            color=""
                            rotate={0}
                            size="normal"
                          >
                            <svg
                              viewBox="0 0 24 24"
                            >
                              <polygon
                                points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
                              />
                            </svg>
                          </div>
                        </IconStyled>
                      </IconSvg>
                    </IconArrowChevronRight>
                  </div>
                </IconWrapperStyled>
              </div>
            </IconExpandWrapperStyled>
          </button>
        </TitleBoxStyled>
      </dt>
    </HeadingStyled>
    <ContentStyled
      aria-labelledby="labelfornull"
      contentBlank={false}
      expanded={false}
      id={null}
      key="content"
      role="region"
    >
      <dd
        aria-labelledby="labelfornull"
        className="ContentStyled-dbeIsI dSVNGC"
        id={null}
        role="region"
      />
    </ContentStyled>
  </AccordionItem>
</WithTheme(AccordionItem)>
`;

exports[`<AccordionItem /> renders correctly with contentBlank prop set to true 1`] = `
Array [
  <dt
    className="HeadingStyled-jcCngC diMfOQ"
    role="heading"
>
    <button
        aria-controls={null}
        aria-expanded={false}
        className="TitleBoxStyled-QGbRI bwRMmk"
        id="labelfornull"
        onClick={[Function]}
        role="button"
    >
        <div
            className="TitleContentStyled-eVpLoT bzPKhr"
        >
            
        </div>
        <div
            className="IconExpandWrapperStyled-jsDvvw cHZrOx"
        >
            <div
                className="IconWrapperStyled-hDFZgP iRWScr"
            >
                <div
                    className="IconSvgStyled-cegYAl gxFYyV"
                    color=""
                    rotate={0}
                    size="normal"
                >
                    <svg
                        viewBox="0 0 24 24"
                    >
                        <polygon
                            points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
                        />
                    </svg>
                </div>
            </div>
        </div>
    </button>
</dt>,
  <dd
    aria-labelledby="labelfornull"
    className="ContentStyled-dbeIsI fgXauG"
    id={null}
    role="region"
/>,
]
`;

exports[`<AccordionItem /> renders correctly with default props 1`] = `
Array [
  <dt
    className="HeadingStyled-jcCngC diMfOQ"
    role="heading"
>
    <button
        aria-controls={null}
        aria-expanded={false}
        className="TitleBoxStyled-QGbRI bwRMmk"
        id="labelfornull"
        onClick={[Function]}
        role="button"
    >
        <div
            className="TitleContentStyled-eVpLoT bzPKhr"
        >
            
        </div>
        <div
            className="IconExpandWrapperStyled-jsDvvw cHZrOx"
        >
            <div
                className="IconWrapperStyled-hDFZgP iRWScr"
            >
                <div
                    className="IconSvgStyled-cegYAl gxFYyV"
                    color=""
                    rotate={0}
                    size="normal"
                >
                    <svg
                        viewBox="0 0 24 24"
                    >
                        <polygon
                            points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
                        />
                    </svg>
                </div>
            </div>
        </div>
    </button>
</dt>,
  <dd
    aria-labelledby="labelfornull"
    className="ContentStyled-dbeIsI dSVNGC"
    id={null}
    role="region"
/>,
]
`;

exports[`<AccordionItem /> renders correctly with expanded id to number 1`] = `
Array [
  <dt
    className="HeadingStyled-jcCngC diMfOQ"
    role="heading"
>
    <button
        aria-controls={1}
        aria-expanded={false}
        className="TitleBoxStyled-QGbRI bwRMmk"
        id="labelfor1"
        onClick={[Function]}
        role="button"
    >
        <div
            className="TitleContentStyled-eVpLoT bzPKhr"
        >
            
        </div>
        <div
            className="IconExpandWrapperStyled-jsDvvw cHZrOx"
        >
            <div
                className="IconWrapperStyled-hDFZgP iRWScr"
            >
                <div
                    className="IconSvgStyled-cegYAl gxFYyV"
                    color=""
                    rotate={0}
                    size="normal"
                >
                    <svg
                        viewBox="0 0 24 24"
                    >
                        <polygon
                            points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
                        />
                    </svg>
                </div>
            </div>
        </div>
    </button>
</dt>,
  <dd
    aria-labelledby="labelfor1"
    className="ContentStyled-dbeIsI dSVNGC"
    id={1}
    role="region"
/>,
]
`;

exports[`<AccordionItem /> renders correctly with expanded id to string 1`] = `
Array [
  <dt
    className="HeadingStyled-jcCngC diMfOQ"
    role="heading"
>
    <button
        aria-controls="testId"
        aria-expanded={false}
        className="TitleBoxStyled-QGbRI bwRMmk"
        id="labelfortestId"
        onClick={[Function]}
        role="button"
    >
        <div
            className="TitleContentStyled-eVpLoT bzPKhr"
        >
            
        </div>
        <div
            className="IconExpandWrapperStyled-jsDvvw cHZrOx"
        >
            <div
                className="IconWrapperStyled-hDFZgP iRWScr"
            >
                <div
                    className="IconSvgStyled-cegYAl gxFYyV"
                    color=""
                    rotate={0}
                    size="normal"
                >
                    <svg
                        viewBox="0 0 24 24"
                    >
                        <polygon
                            points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
                        />
                    </svg>
                </div>
            </div>
        </div>
    </button>
</dt>,
  <dd
    aria-labelledby="labelfortestId"
    className="ContentStyled-dbeIsI dSVNGC"
    id="testId"
    role="region"
/>,
]
`;

exports[`<AccordionItem /> renders correctly with expanded prop set to true 1`] = `
Array [
  <dt
    className="HeadingStyled-jcCngC diMfOQ"
    role="heading"
>
    <button
        aria-controls={null}
        aria-expanded={true}
        className="TitleBoxStyled-QGbRI CKANW"
        id="labelfornull"
        onClick={[Function]}
        role="button"
    >
        <div
            className="TitleContentStyled-eVpLoT bzPKhr"
        >
            
        </div>
        <div
            className="IconExpandWrapperStyled-jsDvvw cHZrOx"
        >
            <div
                className="IconWrapperStyled-hDFZgP kFJCnu"
            >
                <div
                    className="IconSvgStyled-cegYAl gxFYyV"
                    color=""
                    rotate={0}
                    size="normal"
                >
                    <svg
                        viewBox="0 0 24 24"
                    >
                        <polygon
                            points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
                        />
                    </svg>
                </div>
            </div>
        </div>
    </button>
</dt>,
  <dd
    aria-labelledby="labelfornull"
    className="ContentStyled-dbeIsI dtCAup"
    id={null}
    role="region"
/>,
]
`;

exports[`<AccordionItem /> renders correctly with title to string 1`] = `
Array [
  <dt
    className="HeadingStyled-jcCngC diMfOQ"
    role="heading"
>
    <button
        aria-controls={null}
        aria-expanded={false}
        className="TitleBoxStyled-QGbRI bwRMmk"
        id="labelfornull"
        onClick={[Function]}
        role="button"
    >
        <div
            className="TitleContentStyled-eVpLoT bzPKhr"
        >
            <div
                className="TitleStyled-kOmUSD bmgLyZ"
            >
                hello
            </div>
        </div>
        <div
            className="IconExpandWrapperStyled-jsDvvw cHZrOx"
        >
            <div
                className="IconWrapperStyled-hDFZgP iRWScr"
            >
                <div
                    className="IconSvgStyled-cegYAl gxFYyV"
                    color=""
                    rotate={0}
                    size="normal"
                >
                    <svg
                        viewBox="0 0 24 24"
                    >
                        <polygon
                            points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
                        />
                    </svg>
                </div>
            </div>
        </div>
    </button>
</dt>,
  <dd
    aria-labelledby="labelfornull"
    className="ContentStyled-dbeIsI dSVNGC"
    id={null}
    role="region"
/>,
]
`;
